### PR TITLE
(SIMP-8384) svckill killing too many services

### PR DESCRIPTION
**Pull request by jeannegreulich**
- Puppet changed the service provider to return service units whose
  state is static.  Because of this svckill was trying to kill system
  processes.  This change updates the svckill provider to remove
  static services from the kill list.

SIMP-8348 #close

### CODE REVIEW

**Comment by jeannegreulich** [comment]
Things I was considering:
1) calling systemctl is-enabled for each resource might be too many system calls.  Should I just hash the results from the initialize and use that later.  If something is not found in the hash because something has changed from the systemctl in svckill and the Puppet type instances call to get the services should I put it in the kill list or just ignore it.  
2) should I also ignore indirect services because they are spawned from other services.

**Comment by trevor-vaughan** [comment]
@jeannegreulich Yeah, we can't call `is-enabled` for each resource. That's going to slam us completely. Pulling the results from `initialize` into an instance variable should work (I think)

We definitely should ignore indirect services as well.

**Comment by trevor-vaughan** [comment]
@jeannegreulich Did some digging around and I think adding something like the following just before the `return @running_services` statement in the `mode` definition might work:

```ruby
@running_services.delete_if { |x|
  x.last[:provider] &&
  x.last[:provider].respond_to?(:cached_enabled?) &&
  ['static', 'indirect', 'masked', 'bad'].include?(x.last[:provider].respond_to?(:cached_enabled?[:output]))
}
```

File: lib/puppet/provider/svckill/kill.rb [hunk 1]
```diff
@@ -140,16 +140,21 @@ def mode
         end
       elsif [:redhat,:systemd].include?(res[:provider])
         if res[:enable].to_s.eql?("true") || res[:ensure].eql?(:running)
-          @running_services[obj[:name]] = {
-            :provider => obj.provider,
-            :resource => res
-          }
+          #  Only add services that are enabled or disabled.  Killing static or other services can cause
+          #  system errors.
+          if obj.provider.respond_to?(:cached_enabled?) && ['enabled', 'disabled'].include?(obj.provider.cached_enabled?[:output])
```
Comment (trevor-vaughan) [inline lib/puppet/provider/svckill/kill.rb:145]:
```suggestion
          if !obj.provider.respond_to?(:cached_enabled?) || (obj.provider.respond_to?(:cached_enabled?) && ['enabled', 'disabled'].include?(obj.provider.cached_enabled?[:output]))
```

This should cover the non-systemd cases. Good call on pruning prior to adding, much more efficient.

**Comment by jeannegreulich** [comment]
Gitlab tests passed https://gitlab.com/simp/pupmod-simp-svckill/-/pipelines/191791006

**Review by trevor-vaughan — APPROVED** [review]

**Pull request merged**